File: src/problems/leetcode-cn/3314. 构造最小位运算数组 I/solution/python/solution.py
class Solution:
    def minBitwiseArray(self, nums: List[int]) -> List[int]:
        result = []
        for v in nums:
            result += [-1]
            for t in range(0, v):
                if (t | (t + 1)) == v:
                    result[len(result)-1] = t
                    break
        return result

File: src/problems/leetcode-cn/3314. 构造最小位运算数组 I/solution/rust/solution.rs
impl Solution {
    pub fn min_bitwise_array(nums: Vec<i32>) -> Vec<i32> {
        let mut result : Vec<i32> = Vec::new();
        for item in nums {
            result.push(-1);
            for t in 0..item {
                if ((t | (t + 1)) == item) {
                    let mut l = result.len();
                    result[l-1] = t;
                    break;
                }
            }
        }
        return result;
    }
}